components/brave_rewards/core/engine/hash_prefix_store.h

```diff
@@ -0,0 +1,63 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_REWARDS_CORE_ENGINE_HASH_PREFIX_STORE_H_
+#define BRAVE_COMPONENTS_BRAVE_REWARDS_CORE_ENGINE_HASH_PREFIX_STORE_H_
+
+#include <memory>
+#include <string>
+#include <string_view>
+#include <utility>
+
+#include "base/files/file_path.h"
+#include "base/files/memory_mapped_file.h"
+#include "base/sequence_checker.h"
+#include "brave/components/brave_rewards/core/mojom/rewards_database.mojom.h"
+
+namespace brave_rewards {
+
+// Responsible for storage and retrieval of a sorted hash prefix list. The
+// operations of this class will block the current thread on IO.
+class HashPrefixStore : public mojom::HashPrefixStore {
+ public:
+  explicit HashPrefixStore(base::FilePath path);
+  ~HashPrefixStore() override;
+
+  HashPrefixStore(const HashPrefixStore&) = delete;
+  HashPrefixStore& operator=(const HashPrefixStore&) = delete;
+
+  // Opens the hash prefix file, if not already open.
+  bool Open();
```

Comment:
Let's move internal methods to the private section.

Reply:
This method is intentionally public. It's used by the unit tests for instance, but in principle could be used by a client to manually control opening/closing.